### ISSUE TITLE
Cannot use two inputs with the same name

### ISSUE THREAD

**Issue by lucaschen** [react-hook-form/react-hook-form#1212]
**Describe the bug**
I cannot use two inputs with the same name, that I switch between using a checkbox.

**To Reproduce**

```jsx
const { handleSubmit, register, watch } = useForm();

const toggle = watch("toggle");

const onSubmit = data => {
  alert(JSON.stringify(data, null, 2));
};

return (
  <form className="App" onSubmit={handleSubmit(onSubmit)}>
    <input name="toggle" ref={register} type="checkbox" />
    {toggle ? (
      <div>
        <h1>Something here</h1>
        <input name="someValue" ref={register} />
      </div>
    ) : (
      <div>
        <input name="someValue" ref={register} />
        <h1>Something HERE</h1>
      </div>
    )}
    <button type="submit">submit</button>
  </form>
);
```

The `<input name="someValue" />` gets unmounted and re-rendered - the new one doesn't work. Only the original one works (when the checkbox is unchecked).

**Codesandbox link (Required)**
[https://codesandbox.io/s/friendly-chaum-w4tbj](https://codesandbox.io/s/friendly-chaum-w4tbj)

**Expected behavior**
I expect the second input to work just as the first one does.

**Comment by bluebill1049** [comment]
<img width="922" alt="Screen Shot 2020-03-15 at 5 41 18 pm" src="https://user-images.githubusercontent.com/10513364/76696864-366e5b00-66e4-11ea-8d6a-bb02317d6e1e.png">

This is expected behavior, an alternative solution will be unregister the field manually.

**Comment by bluebill1049** [comment]
going to close this issue, feel free to follow up with questions.

**Comment by zelucena** [comment]
I have just found there would be some use to this. I was asked to create two submit buttons: "Save and close, save and create another". I guess I'll have two distinct names and an IF statement. No big deal, just not as elegant.

**Comment by morgs32** [comment]
Or having a different input on mobile vs desktop...

**Comment by SeppeMarien** [comment]
I understand that the name must be unique but what actually happens if it is not? Does the first element get used or the second?

**Comment by bluebill1049** [comment]
> I understand that the name must be unique but what actually happens if it is not? Does the first element get used or the second?

value will overwrite each others.

**Comment by SeppeMarien** [comment]
> > I understand that the name must be unique but what actually happens if it is not? Does the first element get used or the second?
> 
> value will overwrite each others.

Oh okay thank you but I have one more follow up: 

If you one name is set on a div then the value would always be undefined?

**Comment by bluebill1049** [comment]
It's the same as register at useEffect, unless you update its value with setValue

**Comment by NawazKhanMd** [comment]
is this issue resoved?

am facing similar issue ...

**Comment by dbousamra** [comment]
@bluebill1049 This appears fixed in my codebase with 7.20.5. I haven't dug into it, but previously I was unable to have two inputs on the screen with same name. Now it appears to work flawlessly.
